### PR TITLE
add suggestions for API packages

### LINKED ISSUE #474: Add advice on user agents?
Specific to packages wrapping APIs.

Adding an user agent.
Should that user agent be different for package developers themselves and for CI.

### PR DESCRIPTION
**Pull request by maelle**
Fix #474 
Related to #478 
Unsure what should really be in there.

### CODE REVIEW

**Comment by maelle** [comment]
@mpadge could you please review this? I won't merge it before the WIP translation of this chapter is merged.

File: pkg_building.Rmd
Comment (mpadge) [inline pkg_building.Rmd:69]: I suspect a lot of people won't immediately understand "user agent"? Maybe a very brief example might help there?

File: pkg_building.Rmd
Comment (mpadge) [inline pkg_building.Rmd:71]:
```suggestion
- Your package should help with pagination, by allowing the users to not worry about it at all since your package does all necessary requests.
```

**Review by mpadge — CHANGES_REQUESTED** [review]

File: pkg_building.es.Rmd
Comment (yabellini) [inline pkg_building.es.Rmd:71]:
```suggestion
### Comprobación de entradas
```

File: pkg_building.es.Rmd
Comment (yabellini) [inline pkg_building.es.Rmd:73]:
```suggestion
Recomendamos que tu paquete utilice un método coherente de tu elección para [comprobar las entradas](https://blog.r-hub.io/2022/03/10/input-checking/) -- ya sea R base, un paquete de R o ayudantes personalizados.
```

File: pkg_building.es.Rmd
Comment (yabellini) [inline pkg_building.es.Rmd:79]:
```suggestion
- asegúrate de que las peticiones envían un [agente de usuario](https://httr2.r-lib.org/articles/wrapping-apis.html#user-agent) es decir, una forma de identificar qué (tu paquete) o quién envió la solicitud. Se debe poder anular el agente de usuario por defecto del paquete. Lo ideal sería que el agente de usuario fuera diferente en los servicios de integración continua y en los de desarrollo (basándose, por ejemplo, en los nombres de usuario de GitHub de quienes desarrollan).
```

File: pkg_building.es.Rmd
Comment (yabellini) [inline pkg_building.es.Rmd:81]:
```suggestion
- Tu paquete debería ayudar con la paginación, permitiendo que queines lo usen no se preocupen en absoluto de ella, ya que tu paquete realiza todas las peticiones necesarias.
```

File: pkg_building.es.Rmd
Comment (yabellini) [inline pkg_building.es.Rmd:84]:
```suggestion
- Tu paquete podría exportar funciones de alto nivel y funciones de bajo nivel, estas últimas permitiendo a las personas que lo usen llamar directamente a los puntos finales de la API con más control (como `gh::gh()`).
```

File: pkg_building.es.Rmd
Comment (yabellini) [inline pkg_building.es.Rmd:86]:
```suggestion
Para más información, consulta el artículo del blog [Por qué deberías (o no) crear un cliente API](https://ropensci.org/blog/2022/06/16/publicize-api-client-yes-no).
```

**Review by yabellini — COMMENTED** [review]
I doubt how we translate wrapper. It is not in our glossary.  I leave it as was translated here, I remember the discussion, but I need to find the decision.

**Pull request merged**
